general-superstaq/general_superstaq/superstaq_client.py

```diff
@@ -1476,33 +1497,67 @@ def compile(self, json_dict: dict[str, str]) -> dict[str, str]:
+            assert isinstance(job_data["statuses"], list)
+            statuses = job_data["statuses"]
 
         # Exception if any have not been successful
-        if not all(s == _models.CircuitStatus.COMPLETED for s in job_data["statuses"]):
+        if not all(s == gss.models.CircuitStatus.COMPLETED for s in statuses):
             raise gss.SuperstaqException(
                 f"Not all circuits were successfully compiled. Check job ID {job_id} for further "
                 "details."
             )
 
-        # Join circuits together in json string - TODO: make this neater.
-        # Note mypy does not recognize that the above checks ensure there are no None's anywhere,
-        # hence the ignored [arg-type]'s
-        compile_dict = {
-            "initial_logical_to_physicals": "["
-            + ", ".join(str(dico) for dico in job_data["initial_logical_to_physicals"])
-            + "]",
-            "final_logical_to_physicals": "["
-            + ", ".join(str(dico) for dico in job_data["final_logical_to_physicals"])
-            + "]",
-        }
+        assert isinstance(job_data["compiled_circuits"], list)
+        compiled_circuits: list[str] = job_data["compiled_circuits"]
+
+        assert isinstance(job_data["final_logical_to_physicals"], list)
+        final_logical_to_physicals: list[dict[int, int]] = job_data["final_logical_to_physicals"]
+
+        assert isinstance(job_data["initial_logical_to_physicals"], list)
+        initial_logical_to_physicals: list[dict[int, int]] = job_data[
+            "initial_logical_to_physicals"
+        ]
 
-        if circuit_type == _models.CircuitType.CIRQ:
-            compile_dict["cirq_circuits"] = "[" + ", ".join(job_data["compiled_circuits"]) + "]"
-            return compile_dict
+        assert isinstance(job_data["logical_qubits"], list)
+        logical_qubits: list[str] = job_data["logical_qubits"]
 
-        if circuit_type == _models.CircuitType.QISKIT:
-            compile_dict["qiskit_circuits"] = "[" + ", ".join(job_data["compiled_circuits"]) + "]"
-            return compile_dict
+        assert isinstance(job_data["physical_qubits"], list)
+        physical_qubits: list[str] = job_data["physical_qubits"]
+
+        # Join circuits together in json string - TODO: make this neater.
+        if circuit_type == gss.models.CircuitType.QISKIT:
+            qiskit_initial_log_to_phys = [
+                [[k, v] for k, v in q_map.items()] for q_map in initial_logical_to_physicals
+            ]
+            qiskit_final_log_to_phys = [
+                [[k, v] for k, v in q_map.items()] for q_map in final_logical_to_physicals
+            ]
+
+            compile_dict = {
+                "initial_logical_to_physicals": json.dumps(qiskit_initial_log_to_phys),
+                "final_logical_to_physicals": json.dumps(qiskit_final_log_to_phys),
+            }
+            compile_dict["qiskit_circuits"] = "[" + ", ".join(compiled_circuits) + "]"
```

Comment:
it would be nice to get rid of the rest of the manual json string editing. i believe this should work after subsequent suggestions:
```suggestion
            "qiskit_circuits": json.dumps(compiled_circuits),
        }
```